utensor_cgen/backend/utensor/code_generator/rearch/_code_generator.py

```diff
@@ -83,33 +91,60 @@ def _find_required_ops(self, ugraph):
         )
     return ops, placeholders, tensor_var_map
 
-  def _try_get_opt_snippets(self, ugraph, ops, tensor_var_map):
+  def _try_get_opt_snippets(self, ugraph, ops, tensor_var_map, placeholders):
+    # snippets for declaring ops
+    declare_op_snippets = []
+    construct_op_snippets = []
     # snippets to be rendered in the generated function body
     local_snippets = []
     # snippets to be rendered in the weight header file
     weight_snippets = []
+    # inputs/outputs enum names
+    input_enums = []
+    output_enums = []
     # op -> op variable name
     ops_map = {}
     for i, op in enumerate(ops):
       op_var_name = 'op_{:03d}'.format(i)
       ops_map[op] = op_var_name
       # all ops will be declared first
-      declare_snippet = op.get_declare_snippet(op_var_name, tensor_var_map)
+      declare_snippet = op.get_declare_snippet(op_var_name, with_const_params=not self.use_model_api)
```

Comment:
You mentioned this on slack, but how exactly is the `with_const_params` flag being used?

Reply:
declare the ops without construct parameters: for the model api, declare them in the header file.
declare the ops with construct parameters: for the function api, which we need to declare and initialize the op.